Replace log entries using string interpolation with direct arguments

heathbm opened the pull request:
Implementation of #45

SvenAelterman commented:
@heathbm Thank you for your contribution. It's being reviewed now.

One question, are the changes you made comprehensive to resolve the issue across the entire code base, or did you pick and choose specific classes?

heathbm commented:
> @heathbm Thank you for your contribution. It's being reviewed now.
> 
> One question, are the changes you made comprehensive to resolve the issue across the entire code base, or did you pick and choose specific classes?

These are all I could find across the code-base. I added another commit as I missed one. I can't find anymore now searching for `*.Log*`

SvenAelterman commented on src/api/Services/FileSystemOperations.cs:
```diff
@@ -79,8 +79,8 @@ public async Task<Result> CreateFileSystem(string fileSystemName, string owner,
+				string Message = "Error while trying to query for the existence of container '{fileSystemName}' in account '{dlsClientAccountName}': '{exMessage}'.";
+				log.LogError(ex, Message, fileSystemName, dlsClient.AccountName, ex.Message);
 				result.Success = false;
 				result.Message = Message;
```
The message that is returned to the client won't have interpolated strings here.

SvenAelterman commented on src/api/Services/FileSystemOperations.cs:
```diff
@@ -100,8 +100,8 @@ public async Task<Result> CreateFileSystem(string fileSystemName, string owner,
+				string Message = "Error while creating new container '{fileSystemName}' in account '{dlsClientAccountName}': '{exMessage}'.";
+				log.LogError(ex, Message, fileSystemName, dlsClient.AccountName, ex.Message);
 				result.Success = false;
 				result.Message = Message;
```
The message returned to the client wont' have interpolated strings.

SvenAelterman reviewed (changes requested):
Two of the result messages, which are returned the client would contain {variable} instead of being interpolated.

SvenAelterman commented:
@heathbm I think there are two instances where a message returned to the client/caller would not be formatted as expected. Can you address those? It's a bit unfortunate, but it seems like where won't be a way to avoid duplicating the strings.

heathbm commented:
> "Error while trying to query for the existence of container '{fileSystemName}' in account '{dlsClientAccountName}': '{exMessage}'."

@SvenAelterman `string.Format()` should help avoid duplicating strings. However, if you would not like to use it, I can update this to simply use interpolation for the result message.

SvenAelterman commented:
@heathbm I don't have a problem with `string.Format()` but I am unaware that it works with "named" placeholders instead of indexed placeholders.

heathbm commented:
> @heathbm I don't have a problem with `string.Format()` but I am unaware that it works with "named" placeholders instead of indexed placeholders.

@SvenAelterman I updated the string to use indexed placeholders.

SvenAelterman commented:
> > @heathbm I don't have a problem with `string.Format()` but I am unaware that it works with "named" placeholders instead of indexed placeholders.
> 
> @SvenAelterman I updated the string to use indexed placeholders.

My mistake, I just see the updated code now. I'll need to review to confirm that indexed placeholders are going to work for structured logging; I've never tried it or seen it.

heathbm commented:
> > > @heathbm I don't have a problem with `string.Format()` but I am unaware that it works with "named" placeholders instead of indexed placeholders.
> > 
> > 
> > @SvenAelterman I updated the string to use indexed placeholders.
> 
> My mistake, I just see the updated code now. I'll need to review to confirm that indexed placeholders are going to work for structured logging; I've never tried it or seen it.

@SvenAelterman Apologies, I tried it out and unfortunately it does not work so I added a commit to use structed logging and separately string interpolation for the result message.

SvenAelterman commented:
@heathbm Thanks for sticking with it. Approved.
